dask-gateway-server/dask_gateway_server/backends/base.py

```diff
@@ -320,6 +346,68 @@ class ClusterConfig(Configurable):
         config=True,
     )
 
+    cluster_max_memory = MemoryLimit(
+        None,
+        help="""
+        The maximum amount of memory (in bytes) available to this cluster.
+        Allows the following suffixes:
+
+        - K -> Kibibytes
+        - M -> Mebibytes
+        - G -> Gibibytes
+        - T -> Tebibytes
+
+        Set to ``None`` for no limit (default).
```

Comment:
May want to add "Limits may still be imposed by `cluster_max_cores` and `cluster_max_workers`. Similar for the other two